src/llmq/quorums_chainlocks.cpp

```diff
@@ -434,14 +422,34 @@ void CChainLocksHandler::EnforceBestChainLock()
 
             pindex = pindex->pprev;
         }
-        inEnforceBestChainLock = false;
+        // In case blocks from the correct chain are invalid at the moment, reconsider them. The only case where this
+        // can happen right now is when missing superblock triggers caused the main chain to be dismissed first. When
+        // the trigger later appears, this should bring us to the correct chain eventually. Please note that this does
+        // NOT enforce invalid blocks in any way, it just causes re-validation.
+        if (!bestChainLockBlockIndex->IsValid()) {
+            ResetBlockFailureFlags(mapBlockIndex.at(bestChainLockBlockIndex->GetBlockHash()));
+        }
+
+        activateNeeded = chainActive.Tip() != bestChainLockBlockIndex;
```

Comment:
Won't this cause useless activation attempts for cases when we have non-locked tip i.e `chainActive.Tip()->GetAncestor(bestChainLockBlockIndex->nHeight) == bestChainLockBlockIndex`?

Reply:
True, fixed this by checking:
`activateNeeded = chainActive.Tip()->GetAncestor(currentBestChainLockBlockIndex->nHeight) != currentBestChainLockBlockIndex`